src/ai/backend/manager/api/session.py

```diff
@@ -2178,6 +2180,111 @@ async def shutdown_service(request: web.Request, params: Any) -> web.Response:
     return web.Response(status=204)
 
 
+async def find_dependency_sessions(
+    root_session_name_or_id: Union[uuid.UUID, str],
+    db_connection: SAConnection,
+    access_key: AccessKey,
+):
+    dependency_session_cache: Dict[str, List] = {}
```

Comment:
How about using ready-to-use methods, like [aiotools.lru_cache()](https://aiotools.readthedocs.io/en/latest/aiotools.func.html#aiotools.func.lru_cache) instead implementing our own caching method?